### ISSUE TITLE
[Suggestion] Clickable triggers' "click" binds should require actual clicks

### ISSUE THREAD

**Issue by ZwipZwapZapony** [jrbudda/Vivecraft_113#4]
I have "move forward" bound to the trigger of my left HTC Vive Pro/2.0 Controller, and "sprint" bound to the trigger "click" of it. However, I don't have to actually **_click_** the trigger all the way down to start sprinting, I simply have to press it down enough (probably just more than 85-95%?).

Considering that the HTC Vive controllers have "clicking"/"dual-stage" triggers, it feels very odd to me that I don't have to "click" them to sprint, and it makes it very awkward for me to walk at full speed without sprinting in the game.

As such, I would like to suggest that if using a VR controller with a "click-able" trigger, the trigger has to be fully "clicked" down to use the trigger "click" control bindings.

I am using Vivecraft 1.13.2 v1 (revision 0, though revisions 1-3's changelogs don't suggest that this has been done yet).

**Comment by jrbudda** [comment]
If I recall correctly the Vive trigger click is not a microswitch, just a physical detent that is *supposed* to line up with the analog value of the trigger being at 1.0.

**Comment by ZwipZwapZapony** [comment]
If you look at "Case Study 2 - Beat Saber" of VALVe's official ["Rebinding 'Legacy' games for new (VR) controllers"](https://steamcommunity.com/sharedfiles/filedetails/?id=1545634111) guide, it specifically states the following:

_... Some games don't use **the trigger "clicked" bit**, but instead look at the analog value for the trigger and apply some threshold to that to determine if the user has activated the trigger. 
There's nothing wrong with a game doing this, it's just a different way of dealing with a trigger. **The Vive Controller and Knuckles are relatively unusual for even having a click at the end of the trigger.** Oculus Touch, Windows MR, and Xbox 360 controllers only have the analog value to work with. ..._

In other words, Vive controllers' triggers are indeed "clickable".

**Comment by jrbudda** [comment]
The current code in Vivecraft is based on handling the phenomenon described in this thread https://steamcommunity.com/app/358720/discussions/0/385428458163316970/ where the 'pressed' state of the click, coming from steamvr, was still happening before the mechanical click. It was determined by us, as by the folks in that thread, that looking for a value of 1.0 on the axis should relate to the full click. According to that thread there is indeed a microswitch there, but at least when we last looked at it, they were not sending it. It's possible that has changed in the interim but google is not showing any new discussions on the issue.
